src/can.c

```diff
@@ -591,7 +595,23 @@ int thingset_can_init_inst(struct thingset_can *ts_can, const struct device *can
     k_event_init(&ts_can->events);
 
 #ifdef CONFIG_CAN_FD_MODE
-    can_set_mode(ts_can->dev, CAN_MODE_FD);
+    can_mode_t supported_modes;
+    err = can_get_capabilities(can_dev, &supported_modes);
+    if (err == 0 && (supported_modes & CAN_MODE_FD) != 0) {
+        err = can_set_mode(ts_can->dev, CAN_MODE_FD);
+        if (err == 0) {
+            LOG_INF("Enabled CAN-FD mode");
```

Comment:
I'd prefer printing this in debug mode only to avoid too many messages on every start of a device:

```suggestion
            LOG_DBG("Enabled CAN-FD mode");
```

Alternatively, we can add the CAN mode information to the LOG_INF below after address claiming has finished, e.g. like this:

```
LOG_INF("Using CAN node address 0x%.2X with %s", ts_can->node_addr, IS_ENABLED(CONFIG_CAN_FD_MODE) ? "CAN FD" : "classical CAN");
```